features/0746-shorten-url/README.md

```diff
@@ -0,0 +1,167 @@
+| `shorten.oobv2` | Shorten an URL according to the URL shortening rules as defined in the [Out of Band V2 Protocol](https://identity.foundation/didcomm-messaging/spec/#short-url-message-retrieval) from the DIDComm V2 specification. This means the URL shortener MUST adhere to the url shortener rules as defined in this protocol. When this goal code is used, the `url` property in the request message **MUST** include the `_oob` property and the `shortened_url` property in the response message MUST include the `_oobid` property. If shortening an out of band invitation is desired, but doesn't need to follow the shortening rules as defined in the out of band v2 protocol the more generic `shorten` goal code can be used. | [Out of Band V2 Protocol](https://identity.foundation/didcomm-messaging/spec/#short-url-message-retrieval) |
+
+#### Shortened URL
+
+Message sent by the **url-shortener** to the **long-url-provider** to share the shortened url.
+
+```json
+{
+  "@type": "https://didcomm.org/shorten-url/1.0/shortened-url",
+  "@id": "<UUID>",
+  "shortened_url": "<SHORTENED-URL>",
+  "expires_time": <INTEGER>,
+  "~thread": {
+    "thid": "<@id of request-shortened-url>"
+  }
+}
+```
+
+Description of the fields:
+
+- `shortened_url` -- (required) The shortend version of the url
+- `expires_time` -- (optional) Timestamp after which the shortened url is no longer valid. For privacy reasons, the **url-shortener** MUST invalidate the shortend url after the specified timestamp. Follows the semantics of the `_time` property as defined in [RFC 0074: DIDComm Best Practices](../../concepts/0074-didcomm-best-practices/README.md#time). If not defined it means the shortened url will be valid indefinitely (or until the url is invalidated using the `invalidate-shortened-url` message).
+
+The message MUST inclue the `~thread.thid` property with the `@id` value of the [Request Shortened URL](#request-shortened-url) message.
+
+#### Invalidate Shortened URL
+
+Message sent by the **long-url-provider** to the **url-shortener** to invalidate a shortened url. This is useful if the short url is no longer needed and reduces the chance of unwanted exposure. If the **long-url-provider** is authorized to invalidate the shortened url, the **url-shortener** SHOULD invalidate the url due to privacy concerns.
+
+```json
+{
+  "@type": "https://didcomm.org/shorten-url/1.0/invalidate-shortened-url",
+  "@id": "<UUID>",
+  "shortened_url": "<SHORTENED-URL>"
+}
```

Comment:
Perhaps add an expiration for the url.

Something like, this url can be re-used in 90 minutes or 40 days, or 1 year, or never.

The concept is that even though I don't want the link to redirect to me anymore, I don't want someone else using this link and redirecting to someone else's link.